.github/workflows/auto-merge.yml

```diff
@@ -0,0 +1,62 @@
+name: Auto-merge
+
+on:
+  pull_request_target:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  auto-merge:
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    if: github.actor == 'd2ai-bot' || github.actor == 'dependabot[bot]'
+    steps:
+      - name: Generate token
+        uses: actions/create-github-app-token@v2
+        id: app-token
+        with:
+          app-id: ${{ secrets.AUTOMERGE_APP_ID }}
+          private-key: ${{ secrets.AUTOMERGE_PRIVATE_KEY }}
+
+      # Security: Checkout PR code to review before auto-merging
+      # This ensures we're working with the actual PR content
+      - name: Checkout PR
+        uses: actions/checkout@v6
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 1
```

Comment:
Not sure I understand this - isn't this the default behavior of actions/checkout?

Reply:
Also, what's the point of checking out the code at all for this workflow? Nothing from the source is used.